src/Integration.Vsix/SonarLintDaemon/SonarLintDaemon.cs

```diff
@@ -124,5 +138,61 @@ private string GetCmdArgs(int port)
                 " \"--port\" \"{4}\"",
                 jarPath, logPath, InstallationPath, className, port);
         }
+
+        private string CreateTempDirectory()
+        {
+            string tempDirectory = Path.Combine(Path.GetTempPath(), "SonarLintDaemon", Path.GetRandomFileName());
+            Directory.CreateDirectory(tempDirectory);
+            return tempDirectory;
+        }
+
+        public void RequestAnalysis(string path, string charset)
+        {
+            Analyze(path, charset);
+        }
+
+        private async void Analyze(string path, string charset)
+        {
+            var request = new AnalysisReq
+            {
+                BaseDir = path,
+                WorkDir = workingDirectory,
+            };
+            request.File.Add(new InputFile
+            {
+                Path = path,
+                Charset = charset,
+            });
+
+            var channel = new Channel(string.Join(":", DAEMON_HOST, port), ChannelCredentials.Insecure);
```

Comment:
I usually use string.Join to join lists or arrays, to concatenate three strings I would prefer string interpolation or string.Format (first is better in this case IMO):

    $"{DAEMON_HOST}:{port}"